packages/taquito-michelson-encoder/test/tokens/or.spec.ts

```diff
@@ -3,348 +3,1254 @@ import { createToken } from '../../src/tokens/createToken';
+          },
+          {
+            prim: 'or',
+            args: [
+              {
+                prim: 'or',
+                args: [
+                  {
+                    prim: 'pair',
+                    args: [{ prim: 'nat' }, { prim: 'int' }],
+                    annots: ['%myPair'],
+                  },
+                  { prim: 'string', annots: ['%myString'] },
+                ],
+              },
+              { prim: 'mutez', annots: ['%myTez'] },
+            ],
+          },
+        ],
+      },
+      0
+    ) as OrToken;
+    token2LevelOrMixedAnnots = createToken(
+      {
+        prim: 'or',
+        args: [
+          {
+            prim: 'or',
+            args: [{ prim: 'int' }, { prim: 'nat', annots: ['%A'] }],
+          },
+          {
+            prim: 'bool',
+          },
+        ],
+      },
+      0
+    ) as OrToken;
+    token3LevelOrMixedAnnots = createToken(
+      {
+        prim: 'or',
+        args: [
+          {
+            prim: 'bytes',
+          },
+          {
+            prim: 'or',
+            annots: ['A'],
+            args: [
+              {
+                prim: 'or',
+                args: [{ prim: 'int' }, { prim: 'nat' }],
+              },
+              { prim: 'bool' },
+            ],
+          },
+        ],
+      },
+      0
+    ) as OrToken;
+    tokenNestedOrWithoutAnnot = createToken(
+      {
+        prim: 'or',
+        args: [
+          {
+            prim: 'or',
+            args: [
+              { prim: 'or', args: [{ prim: 'address' }, { prim: 'bytes' }] },
+              { prim: 'or', args: [{ prim: 'int' }, { prim: 'nat' }] },
+            ],
+          },
+          {
+            prim: 'or',
+            args: [
+              {
+                prim: 'or',
+                args: [
+                  { prim: 'pair', args: [{ prim: 'nat' }, { prim: 'int' }] },
+                  { prim: 'string' },
+                ],
+              },
+              { prim: 'mutez' },
+            ],
+          },
+        ],
+      },
+      0
+    ) as OrToken;
+    tokenOrWithOption = createToken(
+      {
+        prim: 'or',
+        args: [
+          { prim: 'int' },
+          {
+            prim: 'or',
+            args: [
+              { prim: 'nat' },
+              {
+                prim: 'or',
+                args: [
+                  { prim: 'mutez' },
+                  {
+                    prim: 'option',
+                    args: [
+                      {
+                        prim: 'or',
+                        args: [{ prim: 'int' }, { prim: 'string' }],
+                        annots: [],
+                      },
+                    ],
+                    annots: [],
+                  },
+                ],
+                annots: [],
+              },
+            ],
+            annots: [],
+          },
+        ],
+        annots: [],
+      },
+      0
+    ) as OrToken;
+  });
 
-    describe('EncodeObject', () => {
```

Comment:
can we structure EncodeObject test suite like Encode for readability?
<img width="822" alt="image" src="https://github.com/ecadlabs/taquito/assets/106410553/bf3b805c-bf0f-45e3-8b60-20f972f51c2a">